src/Castle.Core/DynamicProxy/Generators/GeneratorUtil.cs

```diff
@@ -33,13 +35,55 @@ public static class GeneratorUtil
 
 			var arguments = StoreInvocationArgumentsInLocal(emitter, invocation);
 
+			// Extracted from the following `for` loop as an optimization.
+			//
+			// The comparison by name is intentional; any assembly could define that attribute.
+			// See explanation in comment below.
+			Func<object, bool> isIsReadOnlyAttribute =
```

Comment:
Create a local method for this - it will allow compiler to cache the value better (only once, rather than per each invocation).

Reply:
I've run a few experiments to see the compiler automatically cache instantiated delegates, but I haven't been able to observe such behavior (neither in a debug nor release builds). It'll quite literally do what it's told, so the important thing is to take the hidden delegate instantiation out of the loop... it's a quick and easy win. Writing the lambda in-place instead of in a local function or static method produces less line noise. IMO we can suffer that one delegate instantiation per invocation&mdash;if not, we'd possibly have to refactor *a lot* of other code in DynamicProxy!

Reply:
Playing with it more I found this issue: https://github.com/dotnet/roslyn/issues/19431. That's the reason whey are not cached (and this might change in future). If you pass anonymous non-capturing lambda and target e.g. `net462` you'll see the caching in action:
```c#
foreach (var parameter in parameters)
{
  if (parameter.attributes.Any(t => t.FullName == "EEEE"))
  {
    // Do smth
  }
}
```
Compiled:
```c#
foreach (System.ValueTuple<string, Type[]> parameter in parameters)
{
    if (!parameter.Item2.Any(<>c.<>9__0_0 ?? (<>c.<>9__0_0 = <>c.<>9.<CycleWithLocalMethod>b__0_0)))
        continue;
}
```

Later they might enable the similar optimization for the local functions - that's why I suggested it.
Also it looks more natural with local methods. But as this a matter of tastes - it's up to you 😉